modules/fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/fs2/io/Metrics.scala

```diff
@@ -64,21 +66,24 @@ object Metrics {
    * Technically `Resource` doesn't give us much as we don't allocate a thread pool,
    * but it will make sure the last report is issued
    */
-  def resource[F[_]: Sync]: Resource[F, Metrics[F]] =
-    Resource
-      .make(init) { case (res, _) => Sync[F].delay(res.close()) }
-      .map { case (res, reg) => make[F](res, reg) }
-
-  /** Initialise backend resources */
-  def init[F[_]: Sync]: F[(Slf4jReporter, MetricRegistry)] =
-    Sync[F].delay {
-      val registry = new MetricRegistry()
-      val logger = LoggerFactory.getLogger(LoggerName)
-      val reporter = Slf4jReporter.forRegistry(registry).outputTo(logger).build()
-      (reporter, registry)
+  def resource[F[_]: Sync](config: MetricsReporter): Resource[F, Metrics[F]] =
+    for {
+      registry <- Resource.liftF(Sync[F].delay((new MetricRegistry())))
+      rep <- reporter(config, registry)
+    } yield make(rep, registry)
+
+  def reporter[F[_]: Sync](config: MetricsReporter, registry: MetricRegistry): Resource[F, ScheduledReporter] =
```

Comment:
Maybe we could call `report` when closing the `Resource` ? Line 66 says so but I don't see it happening.